### PR TITLE
Editor: Move tests to single test runner

### PR DESCRIPTION
**Pull request by aduth**
Related: #3942

This pull request seeks to move `post-editor` tests to the single test runner, adopting best practices for test setup and teardown as part of the migration.

__Implementation notes:__

Two items worth noting:
- 9790236 changes test file loading to use default require extension resolution, so that `.jsx` test files can be included
- Mockery's `useCacheClean: true` seems to cause some interference with React component testing. Specifically, I was encountering "Only a ReactOwner can have refs" invariant errors until I had moved the `react-addons-test-utils` loading until after Mockery had been enabled. You'll see the errors yourself if you try moving `TestUtils` as a top-level import.

Generally, there were pain points around mocking child components and `this.translate` calls. It may be worth exploring some helpers around these (e.g. move `MOCK_COMPONENT` to `test/helpers`, maybe add helper for injecting `translate` into component(s)), or alternatively find better ways of testing components without needing mocks.

__Testing instructions:__

Ensure tests pass by running `npm run test-client`.

/cc @gziolo, @blowery

### CODE REVIEW

File: client/post-editor/test/post-editor.jsx [hunk 1]
```diff
@@ -1,88 +1,79 @@
...
+		mockery.registerMock( 'react-redux', {
+			connect: () => ( component ) => component
+		} );
 
-	beforeEach( function() {
-		sandbox = sinon.sandbox.create();
+		SitesList = require( 'lib/sites-list/list' );
+		PostEditStore = require( 'lib/posts/post-edit-store' );
+		PostEditor = require( '../post-editor' );
+		PostEditor.prototype.__reactAutoBindMap.translate = ( string ) => string;
```
Comment (blowery) [inline client/post-editor/test/post-editor.jsx]: oof.

File: test/load-suite.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ function requireTestFiles( config, path = '' ) {
 		const folderConfig = config[ folderName ];
 
 		if ( folderName === 'test' ) {
-			folderConfig.forEach( fileName => require( `${path}test/${fileName}.js` ) );
+			folderConfig.forEach( fileName => require( `${path}test/${fileName}` ) );
```
Comment (blowery) [inline test/load-suite.js]: i think this landed on a separate PR...

Comment (blowery) [inline test/load-suite.js]: https://github.com/Automattic/wp-calypso/pull/4061

**Comment by blowery** [comment]
Looks good :ship:

File: test/load-suite.js (hunk 2 above)
Comment (gziolo) [inline test/load-suite.js]: Sometimes git does great job when changes are identical. I've seen similar cases before :100:
